bin/wserver/wserver.ml

```diff
@@ -1,100 +1,188 @@
 (* Copyright (c) 1998-2007 INRIA *)
 
-let connection_closed = ref false
-
-let eprintf = Printf.eprintf
-
-let sock_in = ref "wserver.sin"
-let sock_out = ref "wserver.sou"
+#ifdef WINDOWS
+(* temporary files for process exchange *)
+let in_fname = ref ("~" ^ (Filename.remove_extension (Filename.basename Sys.executable_name)) ^ "_post_rq.txt")
+let out_fname = ref ("~" ^ (Filename.remove_extension (Filename.basename Sys.executable_name)) ^ "_response.txt")
+#endif 
+  
 let stop_server = ref "STOP_SERVER"
 let noproc = ref false
 let cgi = ref false
 
 let wserver_sock = ref Unix.stdout
-let wsocket () = !wserver_sock
-
-let wserver_oc = ref stdout
-let woc () = !wserver_oc
-
-let wflush () = flush !wserver_oc
-
-let skip_possible_remaining_chars fd =
-  let b = Bytes.create 3 in
-  try
-    let rec loop () =
-      match Unix.select [fd] [] [] 5.0 with
-      | [_], [], [] ->
-        let len = Unix.read fd b 0 (Bytes.length b) in
-        if len = Bytes.length b then loop ()
-      | _ -> ()
-    in
-    loop ()
-  with Unix.Unix_error (Unix.ECONNRESET, _, _) -> ()
-
-let close_connection () =
-  if not !connection_closed then begin
-    wflush () ;
-    Unix.shutdown !wserver_sock Unix.SHUTDOWN_SEND ;
-    skip_possible_remaining_chars !wserver_sock ;
-    (* Closing the channel flushes the data and closes the underlying file descriptor *)
-    close_out !wserver_oc ;
-    connection_closed := true
-  end
 
-let printnl () =
-  output_string !wserver_oc "\013\010"
+let shutdown_noerr fd = 
+  try Unix.shutdown fd Unix.SHUTDOWN_ALL with _ -> ()
 
-type printing_state = Nothing | Status | Contents
+type printing_status = Nothing | Status | Contents
 
 let printing_state = ref Nothing
 
+let status_string status = 
+  match status with
+  | Def.OK -> "200 OK"
+  | Def.Moved_Permanently -> "301 Moved Permanently"
+  | Def.Found -> "302 Found"
+  | Def.Bad_Request -> "400 Bad Request"
+  | Def.Unauthorized -> "401 Unauthorized"
+  | Def.Forbidden -> "403 Forbidden"
+  | Def.Not_Found -> "404 Not Found"
+  | Def.Method_Not_Allowed -> "405 Method Not Allowed"
+  | Internal_Server_Error -> "500 Internal Server Error"
+  | Service_Unavailable -> "503 Service Unavailable"
+  | HTTP_Version_Not_Supported -> "505 HTTP Version Not Supported"
+
+(* max_http like TCP PDU size : typical max MTU IP (1400) to MAX TCP PDU (64k) *)
```

Comment:
No this is done deliberately to manually manage sending data. The data goes online using the Unix library (write unbuffered) instead of non-controllable stdlib (error code, behavior according to events on the network, etc ...) 65536 is the normal size of TCP PDU.

Reply:
It may be an enhancement, but it's not a fix, so it can be discussed in a separated PR.

Reply:
it is intrinsically linked to the correction for Windows (no mixing between stdlib / unix library to be able to test error codes via Unix.error. SysError contains a non-testable string)